ModelCatalogueCorePlugin/src/groovy/org/modelcatalogue/core/export/inventory/DataModelToXlsxExporter.groovy

```diff
@@ -0,0 +1,179 @@
+                        style "description"
+                        colspan 2
+                        height 100
+                    }
+                }
+            }
+            row {
+                cell {
+                    value dataModel.combinedVersion
+                    style "model-catalogue-id"
+                    colspan 2
+                }
+            }
+            row {
+                cell {
+                    value dataModel.status
+                    style "status"
+                    colspan 2
+                }
+            }
+            row()
+            row {
+                cell {
+                    value new Date()
+                    style "date"
+                    colspan 2
+                }
+            }
+            row()
+
+            // data classes
+            row {
+                cell {
+                    value "All Inner Data Classes"
+                    style "h2"
+                    colspan 2
+                }
+            }
+
+            row {
+                cell {
+                    value 'ID'
+                    style 'inner-table-header'
+                    width 10
+                }
+                cell {
+                    value 'Name'
+                    style 'inner-table-header'
+                    width 70
+                }
+            }
+            buildDataClassesOutline(it, dataClasses)
+
+            // footer
+            row()
+            row {
+                cell {
+                    value 'Click the data class cell to show the detail'
+                    style 'note'
+                    colspan 2
+                }
+            }
+        }
+    }
+
+    private void buildDataClassesOutline(Sheet sheet, ListWithTotalAndType<DataClass> dataClasses, int level = 1) {
+        dataClasses.items.each { DataClass dataClass ->
+            sheet.row {
+                cell {
+                    value dataClass.combinedVersion
+                    style {
+                        align bottom right
+                    }
+                    link to name DataClassToXlsxExporter.getReferenceName(dataClass)
+                }
+                cell {
+                    value dataClass.name
+                    link to name DataClassToXlsxExporter.getReferenceName(dataClass)
+                    style {
+                        if (level) {
+                            indent (level * 2)
+                        }
+                    }
+                }
+            }
+
+            if (processedDataClasses.containsKey(dataClass.id)) {
+                return
+            }
+
+            if (level > 3) {
+                processedDataClasses.put(dataClass.id, dataClass)
+                return
+            }
+
+            processedDataClasses.put(dataClass.id, dataClass)
+
+            if (dataClass.countParentOf() > 0) {
+                sheet.group {
+                    def children = new ListWithTotalAndType() {
```

Comment:
I don't think I have deep knowledge of it :smile: 

Would be great if we make ListWithTotalAndType implements List or maybe just Iterable. Would this be a perfect use case for a `trait`?